### PR TITLE
fix: useClickAway wrong trigger

### LINKED ISSUE #1250: useClickAway 错误的触发了
来自社区同学的反馈：

> useClickAway 因为内部使用 element.contains 判定，所以，如果容器内的某元素 click 后会导致该元素销毁，也会触发 click away。em... 目前只能通过 stopImmediatePropagation 解决
> https://codesandbox.io/s/recursing-lamarr-w0ib8?file=/src/App.js

### PR DESCRIPTION
**Pull request by jermyn626**
[[中文版模板 / Chinese template](https://github.com/alibaba/hooks/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
fix #1250

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: packages/hooks/src/useClickAway/index.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ export default function useClickAway(
 
   useEffect(() => {
     const handler = (event: any) => {
+      if (!document.contains(event.target)) return;
```
Comment (brickspert) [inline packages/hooks/src/useClickAway/index.ts:19]: target may be array

Comment (jermyn626) [reply packages/hooks/src/useClickAway/index.ts:19]: 这个是事件的 target 呀，这里是判断事件的 target 在不在 dom 树上～

**Comment by brickspert** [comment]
这里不建议修改，有以下几个原因：

1. 如果消失的元素在内部，这么改没问题。但是如果消失的元素在外部，这么改就会有问题。 
2. 建议给内部消失的元素上加个 stoppropagation 来解决这个问题。

**Comment by jermyn626** [comment]
@brickspert 有道理～但是现在问题就在于消失的元素似乎没法通过 DOM api 判断是否在 target 内部。

**Comment by brickspert** [comment]
> @brickspert 有道理～但是现在问题就在于消失的元素似乎没法通过 DOM api 判断是否在 target 内部。

建议加个 stoppropagation 来解决~

**Comment by jermyn626** [comment]
ok～ 那我 close 下

**Pull request closed without merging**